File: csharp/UIGeneration/MultiplePanelNode.cs
using System;
using System.Collections;
using PUC.CIO;

namespace PUC.UIGeneration
{
	/// <summary>
	/// Summary description for MultiplePanelNode.
	/// </summary>
	public abstract class MultiplePanelNode : InterfaceNode
	{
		/*
		 * Member Variables
		 */

		protected ArrayList _panels;


		/*
		 * Constructors
		 */

		public MultiplePanelNode()
		{
			_panels = new ArrayList();
		}


		/*
		 * Properties
		 */

		public ArrayList Panels
		{
			get
			{
				return _panels;
			}
		}


		/*
		 * Member Methods
		 */

		public void AddPanel( InterfaceNode node )
		{
			_panels.Add( node );

			node.SetParent( this );
		}

		public void RemovePanel( InterfaceNode node )
		{
			_panels.Remove( node );

			node.SetParent( null );
		}


		/*
		 * InterfaceNode Methods
		 */

		public override abstract void AddComponents( ContainerCIO container, LayoutVariables vars );

		public override abstract void DoLayout( LayoutVariables vars );

		public override abstract void CalculateMinimumSize(LayoutVariables vars);
		public override abstract void CalculatePreferredSize( LayoutVariables vars );

		public override string ToString()
		{
			string ret = _bounds.ToString() + "\r\n";

			for( int i = 0; i < _panels.Count; i++ )
			{
				ret += i + " . " + _panels[ i ].ToString() + "\r\n";
			}

			return ret;
		}
	}
}

File: edu/cmu/hcii/puc/uigen/MultiplePanelNode.java
/**
 * MultiplePanelNode.java
 *
 * This node represents nodes that branch to multiple panels.
 *
 * @author Jeffrey Nichols
 */

// Package Definition

package edu.cmu.hcii.puc.uigen;


// Import Declarations

import java.awt.Dimension;
import java.awt.Rectangle;

import java.util.Vector;

import edu.cmu.hcii.puc.cio.ContainerCIO;


// Class Definition

public abstract class MultiplePanelNode extends InterfaceNode {

    //**************************
    // Member Variables
    //**************************

    protected Vector m_vPanels;
    

    //**************************
    // Constructor
    //**************************

    public MultiplePanelNode() {
	m_vPanels = new Vector();
    }

    //**************************
    // Member Methods
    //**************************

    public void addPanel( InterfaceNode pNode ) {

	m_vPanels.addElement( pNode );

	pNode.setParent( this );
    }

    public void removePanel( InterfaceNode pNode ) {

	m_vPanels.removeElement( pNode );

	pNode.setParent( null );
    }

    //**************************
    // Interface Methods
    //**************************

    public abstract void addComponents( ContainerCIO pContainer );

    public abstract void doSizing( LayoutVariables pVars );

    public abstract void doLayout( ContainerCIO pContainer, 
				   LayoutVariables pVars );

    public abstract Dimension getPreferredSize( LayoutVariables pVars );

    public String toString() {

	String ret = m_rBounds.toString() + "\n";

	for( int i = 0; i < m_vPanels.size(); i++ ) {
	    
	    ret += i + ". " + m_vPanels.elementAt( i ).toString() + "\n";
	}

	return ret;
    }
}
